extensions/amp-lightbox/1.0/lightbox.js

```diff
@@ -0,0 +1,239 @@
+  ],
+  'fly-in-bottom': [
+    {opacity: 0, transform: 'translate(0, 100%)', visibility: 'visible'},
+    {opacity: 1, transform: 'translate(0, 0)', visibility: 'visible'},
+  ],
+};
+
+// Note: Sometimes, the timing between render and lightboxRef update is delayed, so we have a quick function to wait for the ref element before acting on it.
+/**
+ * @param {Object} ref
+ * @param {number} time
+ * @return {number}
+ */
+function getRefElement(ref, time = 1) {
+  return new Promise((resolve, reject) => {
+    if (time <= 0) {
+      return reject(`${ref} not found`);
+    }
+    if (ref) {
+      return resolve(ref);
+    }
+    setTimeout(() => getRefElement(ref, time - 1).then(resolve), 100);
+  });
+}
+
+/**
+ * @param {!LightboxDef.Props} props
+ * @param {{current: (!LightboxDef.LightboxApi|null)}} ref
+ * @return {PreactDef.Renderable}
+ */
+function LightboxWithRef(
+  {
+    id,
+    animateIn = 'fade-in',
+    closeButtonAriaLabel,
+    children,
+    onOpen,
+    initialOpen,
+    opt_beforeOpen,
+    opt_afterClose,
+    ...rest
+  },
+  ref
+) {
+  // `show` is a trigger variable for animations.
+  // When the user opens/closes the lightbox, `show` is changed, which then triggers the appropriate animation and changes `visible` to true/false
+  // `visible` controls the actual visibility of the lightbox element on the page.
+  // When `visible` is true, the element exists on the page, else, it doesn't.
+  // Two states are necessary here in order to control the timing of the animation properly, else rendering conflicts with animation timing.
+  const [show, setShow] = useState(initialOpen);
+  const [visible, setVisible] = useState(false);
+  const styleClasses = useStyles();
+  const lightboxRef = useRef();
+  const beforeOpen = useCallback(
+    () => (opt_beforeOpen ? opt_beforeOpen() : undefined),
+    [opt_beforeOpen]
+  );
+  const afterClose = useCallback(
+    () => (opt_afterClose ? opt_afterClose() : undefined),
+    [opt_afterClose]
+  );
+
+  // We are using refs here to refer to common strings, objects, and functions used.
+  // This is because they are needed within `useEffect` calls below (but are not depended for triggering)
+  // This bypasses the exhaustive-deps eslint error
+  const onOpenFnRef = useRef();
+  onOpenFnRef.current = onOpen;
+  const animateInRef = useRef();
+  animateInRef.current = animateIn;
+  const lightboxContainWrapperRef = useRef();
+  lightboxContainWrapperRef.current = `.${styleClasses.lightboxContainWrapper}`;
+
+  useImperativeHandle(
+    ref,
+    () => ({
+      open: () => {
+        beforeOpen();
+        setShow(true);
+      },
+      close: () => {
+        setShow(false);
+      },
+    }),
+    [beforeOpen]
+  );
+
+  useEffect(() => {
+    if (show) {
+      setVisible(true);
+    } else {
+      getRefElement(lightboxRef).then((element) => {
+        element = element.current;
+        if (element != undefined) {
+          const animation = element.animate(
```

Comment:
Can animation be disabled?

Reply:
No, in 0.1 animation is always enabled and defaults to `fade-in`. This can definitely be added afterwards though.

Reply:
sg